.gitpod.Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM gitpod/workspace-base
+
+# Installation Prep
+RUN sudo apt-get update && sudo apt-get install -y \
+    libusb-1.0-0-dev \
+    libxshmfence1 \
+    libglu1 \
+    libgtk-3-dev \
+    libgconf2-dev \
+    libxss1 \
+    libnss3-dev \
+    libasound2 \
+    libudev-dev \
+    unzip \
+    cmake
+
+USER gitpod
+
+COPY .nvmrc /home/gitpod/
+RUN NODE_VERSION=$(cat /home/gitpod/.nvmrc) \
+    && echo Node version: $NODE_VERSION \
+    && curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | PROFILE=/dev/null bash \
+    && bash -c ". .nvm/nvm.sh \
+        && nvm install v$NODE_VERSION \
+        && nvm alias default v${NODE_VERSION} \
+        && npm install -g typescript yarn node-gyp \
+        && echo 'export PATH=/home/gitpod/.nvm/versions/node/v${NODE_VERSION}/bin:$PATH' >> /home/gitpod/.bashrc \
+    " \
+    && echo "Node and Yarn installed"
```

Comment:
What do you think about doing something similar in the Companion Docker image (maybe without nvm, but using parts of https://github.com/nodejs/docker-node/blob/main/17/bullseye-slim/Dockerfile https://github.com/nodejs/docker-node/blob/main/17/bullseye/Dockerfile)?

Reply:
Will that save much/any space compared to the `node:14-slim` image we are already using?

Reply:
Nothing or almost nothing, but we can use a specific Node version and we can use less Docker layers, since we can run everything in only one run command.

Reply:
I don't think we need to use a specific version, we've been doing fine with very vague ranges until now. We only need ^14.18 but I don't thin that is valid node-version syntax.

And you will struggle to get less layers than we have without flattening base image layers, which will stop those layers being reused across images. And we need to consider if the fractionally smaller image is worth the maintainable cost

A few weeks ago I did an overhaul of the docker files, as I noticed the image was a massive 3gb. It's down to a sensible size now, going further will be hard.